core/src/main/scala/org/apache/spark/api/python/SerDeUtil.scala

```diff
@@ -153,7 +153,10 @@ private[spark] object SerDeUtil extends Logging {
       iter.flatMap { row =>
         val obj = unpickle.loads(row)
         if (batched) {
-          obj.asInstanceOf[JArrayList[_]].asScala
+          obj match {
+            case array: Array[Any] => array.toList
```

Comment:
I think that `array.toList` constructs a Scala `List` class, which is basically a linked list.  Is there a reason why we can't call `toSeq` instead, or can't simply return the array?